pkg/ring/kv/client.go

```diff
@@ -20,14 +20,22 @@ import (
 var inmemoryStoreInit sync.Once
 var inmemoryStore Client
 
-// Config is config for a KVStore currently used by ring and HA tracker,
-// where store can be consul or inmemory.
-type Config struct {
-	Store      string            `yaml:"store,omitempty"`
+// StoreConfig is a configuration used for building single store client, either
+// Consul, Etcd, Memberlist or MultiClient. It was extracted from Config to keep
+// single-client config separate from final client-config (with all the wrappers)
+type StoreConfig struct {
 	Consul     consul.Config     `yaml:"consul,omitempty"`
 	Etcd       etcd.Config       `yaml:"etcd,omitempty"`
 	Memberlist memberlist.Config `yaml:"memberlist,omitempty"`
-	Prefix     string            `yaml:"prefix,omitempty"`
+	Multi      MultiConfig       `yaml:"multi,omitempty"`
+}
+
+// Config is config for a KVStore currently used by ring and HA tracker,
+// where store can be consul or inmemory.
+type Config struct {
+	Store       string `yaml:"store,omitempty"`
+	Prefix      string `yaml:"prefix,omitempty"`
+	StoreConfig `yaml:",inline"`
 
 	Mock Client
```

Comment:
I known you didn't add it, but this should have `yaml:"-"`, to exclude it from any marshalling/unmarshalling.

Reply:
Done.